ethereum/core/src/test/java/org/hyperledger/besu/ethereum/privacy/PrivateStateRootResolverTest.java

```diff
@@ -0,0 +1,180 @@
+    BLOCKCHAIN = InMemoryStorageProvider.createInMemoryBlockchain(BLOCK_GENERATOR.genesisBlock());
+    for (int i = 1; i <= 69; i++) {
+      final BlockDataGenerator.BlockOptions options =
+          new BlockDataGenerator.BlockOptions()
+              .setBlockNumber(i)
+              .setParentHash(BLOCKCHAIN.getBlockHashByNumber(i - 1).get());
+      final Block block = BLOCK_GENERATOR.block(options);
+      final List<TransactionReceipt> receipts = BLOCK_GENERATOR.receipts(block);
+      BLOCKCHAIN.appendBlock(block, receipts);
+    }
+  }
+
+  @Before
+  public void setUp() {
+    privateStateStorage = InMemoryStorageProvider.createInMemoryPrivateStateStorage();
+  }
+
+  @Test
+  public void mustResolveStateRootIfChainHeadIsNotCommitted() {
+    final BlockDataGenerator.BlockOptions options =
+        new BlockDataGenerator.BlockOptions()
+            .setBlockNumber(BLOCKCHAIN.getChainHeadBlockNumber())
+            .setParentHash(BLOCKCHAIN.getChainHeadHash());
+    final Block block = BLOCK_GENERATOR.block(options);
+    final PrivateStateRootResolver privateStateRootResolver =
+        new PrivateStateRootResolver(privateStateStorage);
+    assertThat(
+            privateStateRootResolver.resolveLastStateRoot(
+                privacyGroupId, block.getHeader().getHash()))
+        .isEqualTo(PrivateStateRootResolver.EMPTY_ROOT_HASH);
+  }
+
+  @Test
+  public void ifNoCommitmentForPrivacyGroupExistsReturnEmptyRootHash() {
+    final PrivateStateRootResolver privateStateRootResolver =
+        new PrivateStateRootResolver(privateStateStorage);
+    assertThat(
+            privateStateRootResolver.resolveLastStateRoot(
+                privacyGroupId, BLOCKCHAIN.getChainHeadHeader().getHash()))
+        .isEqualTo(PrivateStateRootResolver.EMPTY_ROOT_HASH);
+  }
+
+  @Test
+  public void ifCommitmentForPrivacyGroupExistsReturnsRootHash() {
+    final PrivateStateStorage.Updater updater = privateStateStorage.updater();
+    updater.putPrivateBlockMetadata(
+        BLOCKCHAIN.getBlockByNumber(16).get().getHash(),
+        Bytes32.wrap(privacyGroupId),
+        new PrivateBlockMetadata(
+            Collections.singletonList(
+                new PrivateTransactionMetadata(
+                    BLOCK_GENERATOR.transaction().getHash(), pmt1StateHash))));
+    updater.putPrivacyGroupHeadBlockMap(
+        BLOCKCHAIN.getChainHeadHash(),
+        new PrivacyGroupHeadBlockMap(
+            Collections.singletonMap(
+                Bytes32.wrap(privacyGroupId), BLOCKCHAIN.getBlockByNumber(16).get().getHash())));
+    updater.commit();
+    final PrivateStateRootResolver privateStateRootResolver =
+        new PrivateStateRootResolver(privateStateStorage);
+    assertThat(
+            privateStateRootResolver.resolveLastStateRoot(
+                privacyGroupId, BLOCKCHAIN.getChainHeadHash()))
+        .isEqualTo(pmt1StateHash);
+  }
+
+  @Test
+  public void ifMultipleCommitmentsExistsReturnsRootHashForCorrectPrivacyGroup() {
```

Comment:
NIT: `resolveCorrectRootHashWhenMultipleCommitmentsExistForPrivacyGroup` would match the most common naming pattern used for unit tests in Besu.